src/components/useRequests.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 RoadieHQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useEffect, useCallback } from 'react';
+import { useApi } from '@backstage/core';
+import { useAsync, useAsyncFn} from 'react-use';
+import convert from 'xml-js';
+import moment from 'moment';
+import { jiraApiRef } from '../api';
+import { ActivityStreamEntry, ActivityStreamElement, ActivityStreamKeys } from '../types';
+import { AxiosError } from 'axios';
+
+const getPropertyValue = (entry: ActivityStreamEntry, property: ActivityStreamKeys): string => entry[property]?._text;
+const getElapsedTime = (start: string) => moment(start).fromNow();
+const decodeHtml = (html: string) => {
+  const txt = document.createElement("textarea");
+  txt.innerHTML = html;
+  return txt.value;
+};
+
+const handleError = (error: AxiosError) => Promise.reject({
+  message: error?.response?.data?.errorMessages.length && error.response.data.errorMessages[0] || error.request
+});
+
+export const useActivityStream = (size: number) => {
```

Comment:
In terms of structure could we have a hooks folder alongside components and have each hook in its own file ?